3.analyze_data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py

```diff
@@ -119,30 +120,118 @@ def perform_ks_test(_dmso_probs, _treatment_probs):
+    filtered_probadf = pd.merge(probadf, barcode_platemapdf, how="inner", left_on="Metadata_Plate", right_on="Assay_Plate_Barcode")
+
+    filtered_probadf.rename(columns={"Plate_Map_Name": "treatment_type"}, inplace=True)
+
+    # Merge the metadata and platemaps for each type of treatment
+    broaddfs = {ttype: pd.merge(treatment_data[ttype]["metadata"], treatment_data[ttype]["platemap"], how="inner", on="broad_sample") for ttype in treatment_data.keys()}
+
+    # Iterate through each type of treatment
+    for ttype in broaddfs.keys():
+
+        # Rename the treatment columns
+        broaddfs[ttype].rename(columns={treatment_data[ttype]["treatment_column"]: "treatment"}, inplace=True)
+
+        # Set the type of treatment
+        broaddfs[ttype]["treatment_type"] = ttype
+
+        # Change the treatment type data values
+        filtered_probadf["treatment_type"].replace({treatment_data[ttype]["Plate_Map_Name"]: ttype}, inplace=True)
+
+    combined_broaddf = pd.concat(broaddfs.values(), axis=0, ignore_index=True)
+
+    # Combine the probability and treatment data using the well
+    common_broaddf = pd.merge(filtered_probadf, combined_broaddf, how="inner", left_on=["Metadata_Well", "treatment_type"], right_on=["well_position", "treatment_type"])
+
+    # Drop redundant columns for the merge operations
+    common_broaddf.drop(columns=drop_cols, inplace=True)
+
+    return common_broaddf
+
+
+# ## Defining tests and aggregation metric names
+
+# In[8]:
+
+
 # Create a dictionary where the keys represent the name of the comparison or test, and the values are dictionaries
 # the subdictionaries refer to the wrapper function for creating the comparison, and the metric name of the comparison being made
-comp_functions = {"ks_test":
-                  {"statistical_test_function": perform_ks_test,
-                   "comparison_metric": "ks_statistic"}
+comp_functions = {"ks_test":  # Name of the test to perform
+                  {"statistical_test_function": perform_ks_test,  # The function for making comparisons
+                   "comparison_metric": "ks_statistic"}  # The name of the comparison metric
                   }
 
 
-# In[7]:
+# ## Compare treatments and negative controls
+
+# In[9]:
+
+
+# Define columns to group by
+filt_cols = ['Metadata_Plate', 'treatment', 'Metadata_model_type', 'treatment_type', 'Metadata_Well']
+
+phenotype_cols = None
+
+# Store comparison data
+comparisons = None
 
+# Compute comparisons by iterating through each plate
+for proba_path in list(proba_plate_paths):
 
-treatments = sig_test.get_treatment_comparison(comp_functions, treatment_paths, probadf, barcode_platemapdf)
+    # Combine the probability data with the plate manifest files
+    common_broaddf = combine_meta(pd.read_parquet(proba_path))
+
+    # Define the phenotypes only once
+    if phenotype_cols is None:
+        # Define phenotype columns
+        phenotype_cols = common_broaddf.loc[:, "ADCCM":"SmallIrregular"].columns.tolist()
```

Comment:
Consider changing this indexing to be more dynamic. What if one of these classes are missing??? Or if another is added? outside of the range???